### ISSUE TITLE
Use HttpClientFactory instead of direct HttpClient instantiation

### ISSUE THREAD

**Issue by Xeevis** [abpframework/abp#1423]
For every request in CLI there is new **CliHttpClient** instantiated and disposed, which is not proper use.

[From docs](https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests):  
> The original and well-known HttpClient class can be easily used, but in some cases, it isn't being properly used by many developers.

> As a first issue, while this class is disposable, using it with the using statement is not the best choice because even when you dispose HttpClient object, the underlying socket is not immediately released and can cause a serious issue named ‘sockets exhaustion’.

> Therefore, HttpClient is intended to be instantiated once and reused throughout the life of an application."

**HttpClientFactory to the rescue**, HttpClient can be added to DI

```csharp
context.Services.AddHttpClient<CliHttpClient>()
    .ConfigurePrimaryHttpMessageHandler(() => new CliHttpClientHandler());
```

And then injected to the constructor like any other service

```c#
protected HttpClient HttpClient { get; }

NuGetService(
    IJsonSerializer jsonSerializer,
    ICancellationTokenProvider cancellationTokenProvider,
    CliHttpClient cliHttpClient)
{
    JsonSerializer = jsonSerializer;
    CancellationTokenProvider = cancellationTokenProvider;
    HttpClient = cliHttpClient.Client;
}
```
Then it can be just called without disposing
```c#
var response = await HttpClient.GetAsync(url, CancellationTokenProvider.Token)
```

**Comment by Xeevis** [comment]
I'd send PR, but there is regression in `CliHttpClientHandler.cs` that might need to be resolved first from related issue #1373 and it's beyond my understanding of the problem. This line is throwing **System.PlatformNotSupportedException: Operation is not supported on this platform.**
https://github.com/abpframework/abp/blob/c4ced2dea956dc9afadc07b21d8ba6628b0f36ca/framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/Http/CliHttpClientHandler.cs#L10

**Comment by hikalkan** [comment]
already done
